code/modules/ai/action_states/action_state.dm

```diff
@@ -0,0 +1,69 @@
+//A singleton action state; AI components are added to the list and a subsystem will periodically call the process() for this
+//AI components will also GetTargetDir() from the action_state their connected to so they can have a direction to move in
+/datum/element/action_state
+	var/list/ai_components = list() //All AI components currently attached to this action state
+
+/datum/element/action_state/New()
+	..()
+	if(!QDELETED(src)) //No duplicates allowed
+		SSactionstate.action_states += src
+
+/datum/element/action_state/proc/attach_component(datum/component/ai_behavior/ai)
+	if(!QDELETED(ai) && iscarbon(ai.parent))
+		ai_components += ai
```

Comment:
You don't use this and you should not be keeping a reference with nothing attached to the host.

Also as I keep saying you should be attaching to the movable not the component